### ISSUE TITLE
Bad tests for modulo operator

### ISSUE THREAD

**Issue by EmilyBourne** [pyccel/pyccel#1669]
### Describe the bug
The modulo operator is often tested with an expression of the form `mod(arr, arr)`. This expression is susceptible to rounding errors. When such errors occur the result may be incorrect by as much as `arr`. It is therefore a very poor choice of test case. The tests should be updated to use 2 arrays.

### Additional context
Causes many issues with the Intel compiler which seems to use a different algorithm

**Comment by EmilyBourne** [comment]
@ovindu-a  Moving this discussion here.

Yes this issue is free and no-one is working on it. It should be an easy issue to start on as it only involves the tests and therefore doesn't require an in-depth understanding of the code base.

Is the description clear enough or do you need more information?

**Comment by ovindu-a** [comment]
> Is the description clear enough or do you need more information?
>
Yes the description seems fine, I think I will need info on how to set-up etc.

**Comment by EmilyBourne** [comment]
The instructions for getting started are here: https://github.com/pyccel/pyccel/blob/devel/developer_docs/getting_started.md

If you are not on windows you probably just need to do:
```
pip3 install -e --user .[test]
```
from the Pyccel repo.

We use pytest for the tests:  https://docs.pytest.org/en/7.4.x/

**Comment by ovindu-a** [comment]
Thank you, I will get started on it!

**Comment by EmilyBourne** [comment]
> Thank you, I will get started on it!

Don't hesitate to get in touch on the discord if you need any help
